src/util/GlobalChecks.cpp

```diff
@@ -38,6 +58,7 @@ printErrorAndAbort(const char* s1)
 {
     std::fprintf(stderr, "%s\n", s1);
     std::fflush(stderr);
+    printBacktrace();
```

Comment:
Ideally, we should not have to add this to the call sites as this approach would not work for SIGSEGV.

Instead, I think the way to do it is with signal handlers (I *think* that the default c++ handlers that end up calling `std::terminate` also end up calling abort, and therefore SIGABORT).

Easylogging I think almost had it. I tried to fix it, but it doesn't seem to intercept signals properly.

See what I did there in case you can make it work: https://github.com/MonsieurNicolas/stellar-core/tree/backtrace-on-failure

Reply:
btw that branch I linked to is not good (maybe you want to keep the handy "crash" endpoint): we should not be messing with easylogging - I was just trying to re-use what they did there for signal handling, but it doesn't seem to work, maybe because it conflicts with the overrides that asio does...

Reply:
I tried to make this work for several hours but there are a bunch of wrinkles

  - trapping `SIGABRT` itself in asio just .. doesn't seem to work, and I can't tell why; spent a while in a debugger and couldn't figure it out
  - trapping all the _other_ core-dump / fatal signals seems to work but only if tracy is disabled, otherwise it interferes by registering its own handlers
  - either way trapping signals doesn't handle the `releaseAssertOrThrow` case anyways because that unwinds before it calls terminate, losing the backtrace

Overall I've opted to leave something useful if not universal in here. We might want to do a followup later for trapping `SIGSEGV` or something but in general this works for the `releaseAssert` and `releaseAssertOrThrow` cases which I think are worthwhile.